File: src/BKE.Dna.Logger.Host/Reconciliation/DnaReconciliationContract.cs
namespace BKE.Dna.Logger.Host.Reconciliation;

internal static class DnaReconciliationContract
{
    public const string ContractId = "bke-dna-reconciliation-v1";
    public const long StorageWarningBytes = 1_073_741_824L;
    public const string ArchiveScope = "conversation";
    public const bool AutomaticDnaExport = false;
    public const bool AutomaticMarkdownExport = false;
    public const bool MergeSqliteAcrossDevices = false;

    public static bool ShouldNotifyStorage(long workingBytes)
    {
        if (workingBytes < 0)
        {
            throw new ArgumentOutOfRangeException(nameof(workingBytes));
        }

        return workingBytes >= StorageWarningBytes;
    }
}

File: android/app/src/main/kotlin/com/bke/dna/logger/DnaReconciliationContract.kt
package com.bke.dna.logger

object DnaReconciliationContract {
    const val CONTRACT_ID = "bke-dna-reconciliation-v1"
    const val STORAGE_WARNING_BYTES = 1_073_741_824L
    const val ARCHIVE_SCOPE = "conversation"
    const val AUTOMATIC_DNA_EXPORT = false
    const val AUTOMATIC_MARKDOWN_EXPORT = false
    const val MERGE_SQLITE_ACROSS_DEVICES = false

    fun shouldNotifyStorage(workingBytes: Long): Boolean {
        require(workingBytes >= 0) { "workingBytes must be non-negative" }
        return workingBytes >= STORAGE_WARNING_BYTES
    }
}
